tests/test_dockerfile.py

```diff
@@ -0,0 +1,392 @@
+    test_env = {
+            "USER" : "dogecoin",
+            "PATH" : os.environ['PATH'],
+            "DATADIR" : "/dogecoin/.dogecoin",
+            }
+
+    result_args = [
+            abs_path("dogecoind"),
+            "-datadir=/dogecoin/.dogecoin",
+            "-daemon",
+            "-testnet",
+            "-printtoconsole",
+            ]
+    result_env = {
+            "USER" : "dogecoin",
+            "PATH" : os.environ['PATH'],
+            }
+    hook.test(test_args, test_env, result_args, result_env)
+    assert hook.result == hook.reference, hook.error_msg()
+
+    #Mixing arguments with and without values
+    test_args = ["dogecoind", "-daemon", "-maxconnections=150"]
+    test_env = {
+            "USER" : "dogecoin",
+            "PATH" : os.environ['PATH'],
+            "DATADIR" : "/dogecoin/.dogecoin",
+            }
+
+    result_args = [
+            abs_path("dogecoind"),
+            "-datadir=/dogecoin/.dogecoin",
+            "-daemon",
+            "-maxconnections=150",
+            "-printtoconsole",
+            ]
+    result_env = {
+            "USER" : "dogecoin",
+            "PATH" : os.environ['PATH'],
+            }
+    hook.test(test_args, test_env, result_args, result_env)
+    assert hook.result == hook.reference, hook.error_msg()
+
+def test_arguments_double_dash(hook):
+    """Check arguments formates with double-dash like `--testnet`"""
+    test_args = ["dogecoind", "--maxconnections=150", "--paytxfee=0.01"]
+    test_env = {
+            "USER" : "dogecoin",
+            "PATH" : os.environ['PATH'],
+            "DATADIR" : "/dogecoin/.dogecoin",
+            }
+
+    result_args = [
+            abs_path("dogecoind"),
+            "-datadir=/dogecoin/.dogecoin",
+            "--maxconnections=150",
+            "--paytxfee=0.01",
+            "-printtoconsole",
+            ]
+    result_env = {
+            "USER" : "dogecoin",
+            "PATH" : os.environ['PATH'],
+            }
+    hook.test(test_args, test_env, result_args, result_env)
+    assert hook.result == hook.reference, hook.error_msg()
+
+def test_mixing_argument_and_env(hook):
+    """Configure container with arguments and environment variables"""
+    test_args = ["dogecoind", "-maxconnections=150", "-daemon"]
+    test_env = {
+            "USER" : "dogecoin",
+            "PATH" : os.environ['PATH'],
+            "DATADIR" : "/dogecoin/.dogecoin",
+            "TESTNET" : "",
+            }
+
+    result_args = [
+            abs_path("dogecoind"),
+            "-datadir=/dogecoin/.dogecoin",
+            "-maxconnections=150",
+            "-daemon",
+            "-testnet",
+            "-printtoconsole",
+            ]
+    result_env = {
+            "USER" : "dogecoin",
+            "PATH" : os.environ['PATH'],
+            }
+    hook.test(test_args, test_env, result_args, result_env)
+    assert hook.result == hook.reference, hook.error_msg()
+
+def test_equal_argv_and_env(hook):
```

Comment:
Just a question: why is this designed like this?

I'm not entirely sure about the feature. I'd like to know why we do that because I think it can create error situations: if I do `-e PAYTXFEE=1` and add `-paytxfee=0.1`, which one will be used by dogecoind and how do I know that as an operator. Shouldn't we protect the operator against that?

Reply:
I had no idea what should I do in this case, just had the idea to test it. The user still can pass multiple times the same arguments, Dogecoin Core accept it, with or without this environment.

I don't know what happen internally, but the following start the node:
```
./bin/dogecoind -maxtxfee=1 -maxtxfee=2
```
Idk if it should be handled here, or maybe improve https://github.com/dogecoin/dogecoin/issues/1313 ?

Reply:
`dogecoind` takes the last entry. Which in this case means the last arg overwrites the env value.

So if we're at this point not protecting the user (I'm fine with this, just need to make a decision on it) perhaps it makes sense to do a different value assignment for an env and arg of the same type and then test if the resulting order is correct. A rewrite in the logic could easily change this and mess things up, so this would be really good to test.

We can let dogecoin/dogecoin#1313 handle this better for users regardless of how they deploy, yes.

Reply:
> sense to do a different value assignment for an env and arg of the same type and then test if the resulting order is correct

You're speaking about the second assertion of this test ?

Reply:
It can be great to protect this behavior, but yeah maybe it's more appropriate to do handle double configuration inside the Core, make sense to me, and would duplicate verification if it's done here.